lib/view_component/base.rb

```diff
@@ -495,6 +516,14 @@ def sidecar_files(extensions)
         (sidecar_files - [source_location] + sidecar_directory_files + nested_component_files).uniq
       end
 
+      def cache_on(*args)
+        class_eval <<~RUBY, __FILE__, __LINE__ + 1
+          def __vc_cache_args
+            #{args}
```

Comment:
I think it may be better to not just include the args. You could also include the class name of the component to the key of the cache. 

This way, you will not overwrite the cache for the first component with another component that has the same args

FirstComponent.new(foo:, bar:) 
SecondComponent.new(foo: bar:)

Reply:
Also, would you like to add the `updated_at` field if the arguments are model records? It would ensure that the cache is invalidated when you update the associated records.

Reply:
@mikodagatan to get this across the line I will add this in the next pr with docs and other nice to haves